add format file

mforets opened the pull request:
Closes https://github.com/JuliaReach/JuliaReachDevDocs/issues/41

Commit:
add format file

schillic commented on .JuliaFormatter.toml:
```diff
@@ -0,0 +1,12 @@
+short_to_long_function_def = true
+whitespace_in_kwargs = false
+annotate_untyped_fields_with_any = true
+format_docstrings = true
```
Maybe we should reconsider formatting docstrings. It wants to indent lists by two spaces and lines are started early. Since we write the inputs in lists, this would be a big change. Below is an example diff. I do not totally oppose it, but we should be aware of it.

```diff
 ### Input
 
-- `spmexp`  -- sparse matrix exponential
-- `i`       -- row index
-- `backend` -- (optional; default: `get_exponential_backend()`) exponentiation
-               backend
+  - `spmexp`  -- sparse matrix exponential
+  - `i`       -- row index
+  - `backend` -- (optional; default: `get_exponential_backend()`) exponentiation
+    backend
```

mforets replied:
agreed, lets remove it for now

schillic reviewed (commented):
I tried it out and am quite happy except with the docstrings (see below).

I also had a look at some more options we could discuss:
- [`whitespace_ops_in_indices`](https://domluna.github.io/JuliaFormatter.jl/dev/#whitespace_ops_in_indices): `true`
- [alignments](https://domluna.github.io/JuliaFormatter.jl/dev/custom_alignment/)
   - `align_assignment: true`
   - `align_conditional: true`
   - `align_pair_arrow: true`
   - `align_matrix: true`

mforets commented:
interesting, i've added them. im not so sure about what align assignments will produce, but lets try it in LazySets and see. i've added the suggestions.

mforets committed:
Update .JuliaFormatter.toml

schillic commented:
Yes, I also forgot to test them to be honest. Will have a look.

schillic commented:
I did not find any examples for the align options. Three other observations:
- Spaces are removed in type annotations (I tried to avoid that but seems not possible). This is okay for me.
```diff
-function isboundedtype(::Type{<:ExponentialMap{N, S}}) where {N, S}
+function isboundedtype(::Type{<:ExponentialMap{N,S}}) where {N,S}
```
- Strings are aligned (this also happened before adding these align options). Again this is okay for me.
```diff
     @assert length(x) == dim(em) "a vector of length $(length(x)) is " *
-        "incompatible with a set of dimension $(dim(em))"
+                                 "incompatible with a set of dimension $(dim(em))"
```
- The yas style sets `import_to_using` to `true`. Should we change that?
```diff
-import GLPK, JuMP, Random, ReachabilityBase
+using GLPK: GLPK
+using JuMP: JuMP
+using Random: Random
+using ReachabilityBase: ReachabilityBase
```

mforets committed:
Update .JuliaFormatter.toml

schillic reviewed (approved):
I think this is the closest to our style we can get. From my side we can go with it.
We should also update https://juliareach.github.io/JuliaReachDevDocs/latest/guidelines/#Coding-guidelines.
